frontend/its-our-call/src/components/BookmarkStar.tsx

```diff
@@ -11,21 +15,81 @@ interface BookmarkStarProps {
 
 interface BookmarkStarState {
   isBookmarked: boolean;
+  isShowingPopover: boolean;
+  isLoadedSuggestions?: boolean;
+  suggestions: BookmarkSuggestionProps[];
+}
+
+interface BookmarkSuggestionProps {
+  startsBookmarked: boolean;
+  title: string;
+}
+
+interface BookmarkSuggestionState {
+  isSubBookmarked: boolean;
+}
+
+class BookmarkSuggestion extends React.Component<BookmarkSuggestionProps, BookmarkSuggestionState> {
+  constructor(props: BookmarkSuggestionProps) {
+    super(props);
+    this.state = {
+      isSubBookmarked: this.props.startsBookmarked === true,
+    };
+    this.toggleSubBookmark = this.toggleSubBookmark.bind(this);
+  }
+
+  toggleSubBookmark() {
+    this.setState({isSubBookmarked: !this.state.isSubBookmarked});
+    // TODO: prompt other ui callbacks
+  }
+
+  render() {
+    let starIcon = (<IoIosStarOutline />);
+    let bookmarkClass = 'unchecked';
+    if (this.state.isSubBookmarked) {
+      starIcon = (<IoIosStar />);
+      bookmarkClass = 'checked';
+    }
+    return (
+      <div className="suggestion">
+        <div className={'embedded-bookmark-star ' + bookmarkClass} onClick={this.toggleSubBookmark}>{starIcon}</div>
+        <div className="title">{this.props.title}</div>
+      </div>
+    );
+  }
 }
 
 class BookmarkStar extends React.Component<BookmarkStarProps, BookmarkStarState> {
   constructor(props: BookmarkStarProps) {
     super(props);
     this.state = {
       isBookmarked: this.props.isBookmarked === true,
+      isShowingPopover: false,
+      isLoadedSuggestions: true,
+      suggestions: [
+        {startsBookmarked: false, title: 'Example one (Senator, VT)'},
+        {startsBookmarked: false, title: 'Example two (Representative 9th District, VT)'}
+      ],
     };
     this.toggleBookmark = this.toggleBookmark.bind(this);
+    this.onClickOutside = this.onClickOutside.bind(this);
   }
   toggleBookmark() {
     this.setState({isBookmarked: !this.state.isBookmarked});
+    this.setState({isShowingPopover: !this.state.isShowingPopover});
```

Comment:
Since the popover says "bookmarked," I think this should set isShowingPopover to true iff isBookmarked is becoming true.

Reply:
I'm purposefully leaving the ui logic and isBookmarked logic separate for now. We'll fix it after we start getting actual data to bookmark. (Plus it's easier for demoing purposes)